python/hsfs/feature_group.py

```diff
@@ -1684,6 +1683,31 @@ def save(
         # Raises
             `hsfs.client.exceptions.RestAPIError`. Unable to create feature group.
         """
+        if (features is None and len(self._features) > 0) or (
+            isinstance(features, List)
+            and len(features) > 0
+            and isinstance(features[0], feature.Feature)
```

Comment:
```suggestion
            and all([isinstance(f, feature.Feature) for f in features])
```

we might as well check all elements